web/Services/LandlordService.cs

```diff
@@ -100,4 +113,49 @@ public async Task ApproveLandlord(int landlordId,  BricksAndHeartsUser user)
         landlord.ApprovalAdminId = user.Id;
         await _dbContext.SaveChangesAsync();
     }
+
+    public LandlordDbModel? FindLandlordWithInviteLink(string inviteLink)
+    {
+        return _dbContext.Landlords.SingleOrDefault(l => l.InviteLink == inviteLink);
+    }
+    
+    // Link existing landlord with user
+    public async Task<ILandlordService.LinkUserWithLandlordResult> LinkExistingLandlordWithUser(
+        string inviteLink,
+        BricksAndHeartsUser user)
+    {
+        // We want to atomically update multiple records (insert a landlord, then set the user's landlord id), so first start a transaction
+        var transaction = await _dbContext.Database.BeginTransactionAsync();

```

Comment:
```suggestion
        var transaction = await _dbContext.Database.BeginTransactionAsync(IsolationLevel.RepeatableRead);
```

Reply:
ok :-)

Reply:
Hey Ben, just a quick question - I'm trying to write database tests, but so far it doesn't quite work if the method involves a transaction. My code for this method uses a transaction because I reckon that the method resembles RegisterLandlordWithUser much, and so I wrote my code in a similar way to RegisterLandlordWithUser. Do you think that I should modify it in a way so that it doesn't use transactions, so that all of it can actually be tested?

Reply:
Yeah tests with transactions don't work at the moment because of how it's set up 🙁 

I think just don't worry about testing this, you're absolutely right to use a transaction, but it's not worth the effort of setting up testing for it - the tests use transactions to keep each test isolated from all the others, but we can't nest them.

Reply:
right I see - sure thank you!